README.md

```diff
@@ -69,3 +69,63 @@ Invite the webhook bridge to your room (`@_webhook:t2bot.io`) and send the messa
 ```
 
 Format can be `plain` or `html`. Emoji will be converted automatically(`:heart:` becomes ❤); set the `emoji` property to `false` to disable this conversion.
+
+# Running with Docker
+
+There are two gotchas when running with Docker.
+
+First of all the appservice needs to communicate with synapse and there are two ways. Either create a docker network and set the urls accordingly or use public urls (ie webhooks.example.com or example.com:9000). Using `localhost:9000` as the url for appservice won't work because synapse has it's "own" localhost.
+
+Secondly, you need to generate the registration file and let synapse (or whatever server you're using) have access to it. The approach here is to generate it with a temporary container and a [bind mount](https://docs.docker.com/engine/admin/volumes/bind-mounts/) so the file is accessible on the host and can be copied to the homeserver.
+
+Below are a few example of how do achieve this. There is also a `docker-compose.example.yaml`.
+
+**Note** I've had some trouble getting synapse to properly talk to the appservice on the docker network (the reverse works fine). If you experience something similar, expose `9000` on the host or reverse proxy it and use the public url instead of `http://matrix_webhooks_container:9000` in the examples.
```

Comment:
Port 9000 should absolutely never be exposed to the public internet. It's an unencrypted, unsecured, channel for injecting events to the application service.

Reply:
That makes sense. Do you consider a reverse proxy ok or should I remove that part completely?

Still trying to figure out why it not reliably working with docker network both ways.

Reply:
It should be internal-only. Ideally localhost if possible, but otherwise a private network.

Reply:
Alright I figured it out. The hostname of the appservice container in the docker network wasn't valid according to synapse (prolly makes sense)

```
synapse.appservice.api - 228 - WARNING - - push_bulk to http://matrix_webhooks_1:9000/transactions/1 threw exception invalid hostname: matrix_webhooks_1
```

Changing to a valid hostname/container name fixes it and there is no reason to expose `9000` outside the host or even the docker network, which is only accessible to containers connected to it!